src/shared/database/schema.ts

```diff
@@ -7,56 +13,259 @@ export type Generated<T> =
+  kind: string;
+  penumbra_addr: Buffer;
+}
+
+export interface IndexWatermarks {
+  height: Int8;
+  index_name: string;
+}
+
+export interface InsightsShieldedPool {
+  asset_id: Buffer;
+  current_value: string;
+  height: Int8;
+  total_value: string;
+  unique_depositors: number;
+}
+
+export interface InsightsShieldedPoolLatest {
+  asset_id: Buffer | null;
+  current_value: string | null;
+  height: Int8 | null;
+  total_value: string | null;
+  unique_depositors: number | null;
+}
+
+export interface InsightsSupply {
+  height: Int8;
+  market_cap: Generated<number | null>;
+  price: number | null;
+  price_numeraire_asset_id: Buffer | null;
+  staked: Int8;
+  total: Int8;
+}
+
+export interface StakeDelegationTxs {
+  amount: Int8;
+  height: Int8;
+  id: Generated<number>;
+  ik: string;
+  tx_hash: Buffer;
+}
+
+export interface StakeSlashings {
+  epoch_index: Int8;
+  height: Int8;
+  id: Generated<number>;
+  ik: string;
+  penalty: string;
+}
+
+export interface StakeUndelegationTxs {
+  amount: Int8;
+  height: Int8;
+  id: Generated<number>;
+  ik: string;
+  tx_hash: Buffer;
+}
+
+export interface StakeValidatorSet {
+  bonding_state: string;
+  definition: string;
+  id: Generated<number>;
+  ik: string;
+  name: string;
+  queued_delegations: Int8;
+  queued_undelegations: Int8;
+  validator_state: string;
+  voting_power: Int8;
+}
+
+export interface SupplyTotalStaked {
+  del_um: Int8;
+  height: Int8;
+  rate_bps2: Int8;
+  um: Int8;
+  validator_id: number;
+}
+
+export interface SupplyTotalUnstaked {
+  arb: Int8;
+  auction: Int8;
+  dex: Int8;
+  fees: Int8;
+  height: Int8;
+  um: Int8;
+}
+
+export interface SupplyValidators {
+  id: Generated<number>;
+  identity_key: string;
+}
+
+interface RawDB {
+  _insights_shielded_pool_depositors: _InsightsShieldedPoolDepositors;
+  _insights_validators: _InsightsValidators;
+  block_details: BlockDetails;
+  dex_ex_aggregate_summary: DexExAggregateSummary;
+  dex_ex_pairs_block_snapshot: DexExPairsBlockSnapshot;
+  dex_ex_pairs_summary: DexExPairsSummary;
   dex_ex_price_charts: DexExPriceCharts;
-  dex_ex_summary: DexExSummary;
+  governance_delegator_votes: GovernanceDelegatorVotes;
+  governance_proposals: GovernanceProposals;
+  governance_validator_votes: GovernanceValidatorVotes;
+  ibc_transfer: IbcTransfer;
+  index_watermarks: IndexWatermarks;
+  insights_shielded_pool: InsightsShieldedPool;
+  insights_shielded_pool_latest: InsightsShieldedPoolLatest;
+  insights_supply: InsightsSupply;
+  stake_delegation_txs: StakeDelegationTxs;
+  stake_slashings: StakeSlashings;
+  stake_undelegation_txs: StakeUndelegationTxs;
+  stake_validator_set: StakeValidatorSet;
+  supply_total_staked: SupplyTotalStaked;
+  supply_total_unstaked: SupplyTotalUnstaked;
+  supply_validators: SupplyValidators;
 }
+
+export type DB = Pick<
+  RawDB,
+  | 'dex_ex_aggregate_summary'
+  | 'dex_ex_pairs_block_snapshot'
+  | 'dex_ex_pairs_summary'
+  | 'dex_ex_price_charts'
+>;
```

Comment:
Previously relevant: https://github.com/penumbra-zone/dex-explorer/pull/123#pullrequestreview-2410934582 cc @cronokirby

We can't both have an auto schema generator and not include some irrelevant types. Hoping with `scripts/generate-pindexer-schema.ts` and this `Pick` it is a fair compromise given developers will not be able to query outside this list.